### DISCUSSION TITLE
Script for automatically hoisting note

### DISCUSSION THREAD

**Discussion by Mangiola** [TriliumNext/Trilium#1361] [Q&A]
I'm attempting to make a script that automatically hoists a note when it is selected. I'm able to achieve this with the following line: `api.setHoistedNoteId(api.getActiveTabNote().noteId);`

One problem: unhoisting a note while viewing its child will bring focus back to the hoisted note. The script then runs again, causing the note to immediately hoist itself a second time. The only way to actually unhoist the note is to do it while the parent note is being displayed.

![demo](https://user-images.githubusercontent.com/24983951/97744000-de426880-1abc-11eb-90c2-0164464b32a4.gif)

Ideally, I would have the script only run if the previously selected note is not its child. I scanned the API docs for a workaround but I'm stumped. Any suggestions are appreciated!

**Comment by avinashkanaujiya** [comment]
Which version are you using?
And what label or relation did you used on Parent to make it work?
I'm trying to replicate it.

**Reply by Mangiola** [reply]
I'm using 0.45.2, though this behavior was also present when I tried it in 0.44.
The parent has the renderNote relation and points to a note called Script (the JS code snippet seen in the gif)

**Reply by avinashkanaujiya** [reply]
It is not working in my setup. I will try it again tommorrow.

**Reply by avinashkanaujiya** [reply]
I mean code is working. And when I'am clicking on the play button on top right, it is hoisting the note. But when I am adding renderNote to a target note and pointing to script, it is not working.
